mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -133,13 +133,13 @@ public void onFailure(Call call, int statusCode, Headers headers, byte[] respons
                 message = getString(R.string.error_unable_to_resolve_hostname);
             } else if (error instanceof SSLHandshakeException) {
                 // if ssl exception, check for some common problems
-                if (error.getCause() instanceof CertPathValidatorException) {
+                if (error.getCause().getCause() instanceof CertPathValidatorException) {
                     message = getString(R.string.error_certificate_not_trusted);
-                } else if (error.getCause() instanceof CertificateExpiredException) {
+                } else if (error.getCause().getCause() instanceof CertificateExpiredException) {
```

Comment:
When testing against https://expired.badssl.com the CertificateExpiredException is even one level under this:
e.getCause().getCause().getCause()

However, don't we've a better way of doing it? E.g. having a loop which calls getCause() until the returned object is not the same as the current one and then using this one for these checks, here?

Reply:
The first implementation had .toString.equals(CertificateExpiredException). I'm not sure if thats better.